### PR TITLE
convert disposal unit to TGUI

### PR DESCRIPTION
**Pull request by intercepti0n**
## About The Pull Request

A bit tweaked disposal units code to make it work with new UI. Converted its UI to TGUI, nothing special.

## Why It's Good For The Game

New UI is more responsive to things happening inside a unit, also it is more colorful and has a progress bar...

## Testing

It builds okay, buttons in UI are workig as intended. 

<details>
<summary>Screenshots</summary>

vts8MepDpr
b2KvZWL9Nr
Pzg1j22THs

</details>

## Changelog
:cl:
refactor: converted Disposal Unit from browser to TGUI.
/:cl:

### CODE REVIEW

**Comment by thevandie** [comment]
so long as this ACTUALLY works, THANK YOU! please do more!

File: code/modules/recycling/disposal.dm [hunk 1]
```diff
@@ -74,46 +78,49 @@
...
-						mode=-1 // Set it to doubleoff l0l
-						to_chat(user, "You remove the screws around the power connection.")
-						return
-					else if(mode==-1)
-						mode=0
-						to_chat(user, "You attach the screws around the power connection.")
-						return
+
+			if(mode != DISPOSALS_OFF)
+				return
```
Comment (hyperioo) [inline code/modules/recycling/disposal.dm:96]:
```suggestion
			if(mode != DISPOSALS_OFF)
				to_chat(user, "Turn off the pump first!")
				return
```

File: code/modules/recycling/disposal.dm [hunk 2]
```diff
@@ -74,46 +78,49 @@
...
-					var/obj/structure/disposalconstruct/C = new (src.loc)
-					src.transfer_fingerprints_to(C)
-					C.pipe_type = PIPE_TYPE_BIN
-					C.anchored = TRUE
-					C.density = TRUE
-					C.update()
-					qdel(src)
+
+			if(!panel_open || mode != DISPOSALS_OFF)
+				return
```
Comment (hyperioo) [inline code/modules/recycling/disposal.dm:112]:
```suggestion
			if(!panel_open || mode != DISPOSALS_OFF)
			    to_chat(user, "You cannot work on the disposal unit if it is not turned off with its power connection exposed.")
				return
```

File: code/modules/recycling/sortingmachinery.dm [hunk 3]
```diff
@@ -448,33 +448,35 @@
...
-						c_mode=1 // Set it to doubleoff l0l
-						to_chat(user, "You remove the screws around the power connection.")
-						return
-					else if(c_mode==1)
-						c_mode=0
-						to_chat(user, "You attach the screws around the power connection.")
-						return
+
+			if(mode != DISPOSALS_OFF)
+				return
```
Comment (hyperioo) [inline code/modules/recycling/sortingmachinery.dm:456]:
```suggestion
			if(mode != DISPOSALS_OFF)
			        to_chat(user, "Turn off the pump first!")
				return
```

File: code/modules/recycling/sortingmachinery.dm [hunk 4]
```diff
@@ -448,33 +448,35 @@
...
-					to_chat(user, "You sliced the floorweld off the disposal unit.")
-					var/obj/structure/disposalconstruct/C = new (src.loc)
-					src.transfer_fingerprints_to(C)
-					C.pipe_type = PIPE_TYPE_INTAKE
-					C.anchored = TRUE
-					C.density = TRUE
-					C.update()
-					qdel(src)
+			if(!panel_open || mode != DISPOSALS_OFF)
+				return
```
Comment (hyperioo) [inline code/modules/recycling/sortingmachinery.dm:468]:
```suggestion
			if(!panel_open || mode != DISPOSALS_OFF)
			        to_chat(user, "You cannot work on the delivery chute if it is not turned off with its power connection exposed.")
				return
```

File: code/modules/recycling/sortingmachinery.dm [hunk 5]
```diff
@@ -448,33 +448,35 @@
...
+
+			if(I.use_tool(user, src, WORKTIME_NORMAL, tool_type, FAILCHANCE_VERY_EASY))
+				to_chat(user, "You sliced the floorweld off the disposal unit.")
```
Comment (hyperioo) [inline code/modules/recycling/sortingmachinery.dm:471]:
```suggestion
				to_chat(user, "You sliced the floorweld off the delivery chute.")
```

**Review by hyperioo — APPROVED** [review]

File: code/modules/recycling/sortingmachinery.dm [hunk 6]
```diff
@@ -448,33 +448,37 @@
...
-					var/obj/structure/disposalconstruct/C = new (src.loc)
-					src.transfer_fingerprints_to(C)
-					C.pipe_type = PIPE_TYPE_INTAKE
-					C.anchored = TRUE
-					C.density = TRUE
-					C.update()
-					qdel(src)
+			if(!panel_open || mode != DISPOSALS_OFF)
+			        to_chat(user, "You cannot work on the delivery chute if it is not turned off with its power connection exposed.")
+				return
```
Comment (hyperioo) [inline code/modules/recycling/sortingmachinery.dm:470]:
```suggestion
			if(!panel_open || mode != DISPOSALS_OFF)
				to_chat(user, "You cannot work on the delivery chute if it is not turned off with its power connection exposed.")
				return
```

File: code/modules/recycling/sortingmachinery.dm [hunk 7]
```diff
@@ -448,33 +448,37 @@
...
-						c_mode=1 // Set it to doubleoff l0l
-						to_chat(user, "You remove the screws around the power connection.")
-						return
-					else if(c_mode==1)
-						c_mode=0
-						to_chat(user, "You attach the screws around the power connection.")
-						return
+
+			if(mode != DISPOSALS_OFF)
+			        to_chat(user, "Turn off the pump first!")
```
Comment (hyperioo) [inline code/modules/recycling/sortingmachinery.dm:456]:
```suggestion
			to_chat(user, "Turn off the pump first!")
```

File: code/modules/recycling/sortingmachinery.dm [hunk 8]
```diff
@@ -448,33 +448,37 @@
...
-						to_chat(user, "You remove the screws around the power connection.")
-						return
-					else if(c_mode==1)
-						c_mode=0
-						to_chat(user, "You attach the screws around the power connection.")
-						return
+
+			if(mode != DISPOSALS_OFF)
+			to_chat(user, "Turn off the pump first!")
+				return
```
Comment (hyperioo) [inline code/modules/recycling/sortingmachinery.dm:457]:
```suggestion
			if(mode != DISPOSALS_OFF)
				to_chat(user, "Turn off the pump first!")
				return
```

File: code/modules/recycling/disposal.dm [hunk 9]
```diff
@@ -74,46 +78,51 @@
...
-					src.transfer_fingerprints_to(C)
-					C.pipe_type = PIPE_TYPE_BIN
-					C.anchored = TRUE
-					C.density = TRUE
-					C.update()
-					qdel(src)
+
+			if(!panel_open || mode != DISPOSALS_OFF)
+			    to_chat(user, "You cannot work on the disposal unit if it is not turned off with its power connection exposed.")
+				return
```
Comment (hyperioo) [inline code/modules/recycling/disposal.dm:114]:
```suggestion
			if(!panel_open || mode != DISPOSALS_OFF)
				to_chat(user, "You cannot work on the disposal unit if it is not turned off with its power connection exposed.")
				return
```

**Pull request merged**